lib/LaTeXML/Core/Gullet.pm

```diff
@@ -254,15 +255,13 @@ sub readXToken {
     if (!defined $token) {
       return unless $$self{autoclose} && $toplevel && @{ $$self{mouthstack} };
       $self->closeMouth; }    # Next input stream.
-    elsif (my $unexpanded = $token->is_notexpanded) {
-      # Should only occur IMMEDIATELY after expanding \noexpand (by readXToken),
-      # so this token should never leak out through an EXTERNAL call to readToken.
-      return $unexpanded; }    # Just return the next token.
     elsif (($cc = $$token[1]) == CC_COMMENT) {    # NOTE: Inlined ->getCatcode
       return $token if $commentsok;
       push(@{ $$self{pending_comments} }, $token); }    # What to do with comments???
     elsif ($cc == CC_MARKER) {
       LaTeXML::Core::Definition::stopProfiling($token, 'expand'); }
+    elsif ($$token[2]) {                                # Inline is_dont_expand
+      return $token; }
```

Comment:
Kinda bizarre that this still works (if it does); the whole point is if you're forced to expand a dont-expand token, you should get the *normal* token. Otherwise, it will never expand.

Reply:
Finicky to think and talk about, let me try.

1. I _think_ that we want to get the token returned by `readXToken` marked as `dont_expand` when it was returned by that route, so that the requesting bit of logic, say `\ifcat` can be looking at it flagged appropriately and have the right comparison outcome.

2. A token is just a (string, catcode) pair, so returning `$token` or `$$token[2]` with your technique is functionally equivalent for all other TeX purposes. You still end up looking up the same meaning/definition, having the same effects, etc etc.

Reply:
But readXToken at ```\noexpand\something``` will return the dont-expand token to be seen by the ifs.  OTOH, returning the still-dont-expand token at this point will keep the token from *ever* expanding!

Reply:
OBTW: Storing the normal token as a secret 3rd element of the dont-expand token is "Magic"; open-coding it makes it "Black".  Jus' sayin'

Reply:
Ah, I think you may be right now that I think this through with more token. I'm working on a test case to exercise reusing a noexpand token now...

Reply:
I added a test which failed with my `return $token` that passes as in the PDF with returning the inner token. Thanks for catching that. Today I feel like I am probably better of just adding new tests than thinking the details through in my head, keep getting tripped up.